File: solutions/csharp/log-levels/1/LogLevels.cs
static class LogLine
{
    public static string Message(string logLine)
    {
        var position = logLine.IndexOf(":") + 1;
        string result = logLine.Substring(position);
        return result.Trim();
    }

    public static string LogLevel(string logLine)
    {
        var begin = logLine.IndexOf("[") + 1;
        var end = logLine.IndexOf("]") - 1;
        string result = logLine.Substring(begin, end);
        return result.ToLower();
    }

    public static string Reformat(string logLine)
    {
        return Message(logLine) + " (" + LogLevel(logLine) + ")";
    }
}

File: solutions/cpp/log-levels/1/log_levels.cpp
//#include <format> //C++20 not supported... :(
#include <string>

namespace log_line
{
    [[nodiscard]] std::string message( const std::string line )
    {
        size_t position{ line.find( ':' ) + 2 };
        return line.substr( position );
    }

    [[nodiscard]] std::string log_level( const std::string line )
    {
        size_t startPosition{ line.find( '[' ) + 1 };
        size_t endPosition{ line.find( ']' ) };
        size_t length{ endPosition - startPosition };
        return line.substr( startPosition, length );
    }

    [[nodiscard]] std::string reformat( const std::string line )
    {
        //return std::format( "{} ({})", message( line ), log_level( line ) );
        return message( line ) + " (" + log_level( line ) + ")";
    }
}
